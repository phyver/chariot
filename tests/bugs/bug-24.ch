data tree('a,'x) where
    | Branch :  'x -> ('a -> tree('a,'x)) -> tree('a,'x)

val
  | f (Branch x b) = Branch x $ g b
and
  | g b x = f ???

:show f
:show g

-- there was a problem when composing the Angel, giving an empty approximation...
