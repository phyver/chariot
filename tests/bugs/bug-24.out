>> option verbose is set to 1
>> inductive type tree was succesfully defined
>> Typing for f, g successful
>> the definition for f is complete
>> the definition for g is complete
>> the definitions f, g are provably correct

val
    f : tree('a,'b) → tree('c,'b)
  | f (Branch⁰ x b) = Branch⁰ x (g b)

val
    g : ('a → tree('a,'b)) → 'c → tree('c,'b)
  | g b x = f ✠

